src/draggable.ts

```diff
@@ -84,14 +160,28 @@ export class Draggable extends EventTarget {
       this.width = this.options.width || $parent.width;
       this.height = this.options.height || $parent.height;
     }
+    const topLeft = new Point($parent.bounds.x - this.$el.bounds.x, $parent.bounds.y - this.$el.bounds.y);
+    const bottomRight = new Point(topLeft.x + this.width, topLeft.y + this.height);
+    this.areaBounds = {
+      topLeft,
+      bottomRight,
+    };
   }
 
   /** Sets the position of the element. */
   setPosition(x: number, y: number) {
-    const m = this.options.margin || 0;
+    // const m = this.options.margin || 0;
 
     let p = new Point(this.options.moveX ? x : 0, this.options.moveY ? y : 0)
-        .clamp(new Bounds(0, this.width, 0, this.height), m)
+        /* .clamp(
+            new Bounds(
+                this.areaBounds.topLeft.x,
+                this.areaBounds.bottomRight.x,
+                this.areaBounds.topLeft.y,
+                this.areaBounds.bottomRight.y
+            ),
+            m
+        )*/
```

Comment:
We need to fix this before merging, for example using another configuration option